magnum_cluster_api/resources.py

```diff
@@ -419,6 +419,19 @@ def get_object(self) -> objects.KubeadmControlPlaneTemplate:
                     "template": {
                         "spec": {
                             "kubeadmConfigSpec": {
+                                "format": "ignition",
+                                "ignition": {
+                                    "containerLinuxConfig": {
+                                        "additionalConfig": "storage:\n  links:\n  - path: /etc/systemd/system/kubeadm.service.wants/containerd.service\n    target: /usr/lib/systemd/system/containerd.service\n"
+                                    },
+                                },
+                                # In Flatcar kubeadm configuration is in different directory because /run
+                                # can't be provisioned with ignition.
+                                "preKubeadmCommands": [
+                                    """
+                                    bash -c "sed -i 's/__REPLACE_NODE_NAME__/$(hostname -s)/g' /etc/kubeadm.yml"
+                                    """
```

Comment:
why couldn't we use `{{ local_hostname }}` -- does that not work with ignition or?

Reply:
`{{ local_hostname }}` is instance metadata from cloud-init.
Flatcar ignition doesn't support such metadatas.
Btw, Flatcar Container Linux ships with a small utility, coreos-metadata, which fetches information specific to the environment in which Flatcar Container Linux is running.  https://github.com/flatcar/afterburn/blob/master/docs/container-linux-legacy.md
I will use `${COREOS_OPENSTACK_HOSTNAME}` instead.